### PR TITLE
EVEREST-173: resource usage endpoint

### PR DESCRIPTION
**Pull request by michal-kralik**
- Returns available and allocated resources per cluster
- Resource allocation functionality has been ported from PMM

### CODE REVIEW

**Review by gen1us2k — CHANGES_REQUESTED** [review]
Thanks for the implementation. However, it misses API tests

**Review by gen1us2k — APPROVED** [review]

**Comment by michal-kralik** [comment]
The test shall pass after https://github.com/percona/percona-everest-cli/pull/72 is merged

File: pkg/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -75,7 +91,21 @@ func (k *Kubernetes) ClusterName() string {
...
+	if err != nil {
+		return ClusterTypeUnknown, err
+	}
+	for _, storageClass := range storageClasses.Items {
+		if strings.Contains(storageClass.Provisioner, "aws") {
+			return ClusterTypeEKS, nil
+		}
+		if strings.Contains(storageClass.Provisioner, "minikube") ||
+			strings.Contains(storageClass.Provisioner, "kubevirt.io/hostpath-provisioner") ||
+			strings.Contains(storageClass.Provisioner, "standard") {
```
Comment (recharte) [inline pkg/kubernetes/kubernetes.go:106]: I don't think this is necessarily correct but we can fix this when we support other k8s cluster types.

**Review by recharte — APPROVED** [review]
I'm not really sure that this custom logic for each k8s cluster type is entirely correct but it was working for PMM and I think it serves our current needs.

File: pkg/kubernetes/kubernetes.go (hunk 1 above)
Comment (michal-kralik) [reply pkg/kubernetes/kubernetes.go:106]: Me neither. I agree this may need a review.

**Pull request merged**